// evergreen-ui/esm/tabs/src/SidebarTab.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React, { PureComponent } from 'react';
import Box from 'ui-box';
import Tab from './Tab';

var SidebarTab =
/*#__PURE__*/
function (_PureComponent) {
  _inherits(SidebarTab, _PureComponent);

  function SidebarTab() {
    _classCallCheck(this, SidebarTab);

    return _possibleConstructorReturn(this, _getPrototypeOf(SidebarTab).apply(this, arguments));
  }

  _createClass(SidebarTab, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          height = _this$props.height,
          isSelected = _this$props.isSelected,
          props = _objectWithoutProperties(_this$props, ["children", "height", "isSelected"]);

      return React.createElement(Tab, _extends({
        isSelected: isSelected,
        height: height
      }, SidebarTab.styles, props, {
        display: "flex"
      }), React.createElement(Box, {
        is: "span",
        flex: "1"
      }, children));
    }
  }]);

  return SidebarTab;
}(PureComponent);

SidebarTab.displayName = "SidebarTab";

_defineProperty(SidebarTab, "propTypes", _objectSpread({}, Tab.propTypes));

_defineProperty(SidebarTab, "defaultProps", {
  height: 32
});

_defineProperty(SidebarTab, "styles", {
  width: '100%',
  paddingX: 0,
  paddingLeft: 8,
  marginX: 0,
  marginBottom: 4,
  justifyContent: 'auto'
});

export { SidebarTab as default };
