// evergreen-ui/commonjs/theme/src/default-theme/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _foundationalStyles = require("./foundational-styles");

var _typography = require("./typography");

var _componentSpecific = require("./component-specific");

var _themeHelpers = require("./theme-helpers");

/**
 * Theme
 * ---
 * The theme object is used to style Evergreen.
 * It is passed into the  `<ThemeProvider theme={theme} />`.
 * ----
 * You can use this as a template for your own themes.
 */

/**
 * Foundational Styles.
 * ---
 * The following properties are NOT REQUIRED by Evergreen.
 * It's exposed for convenience and documentation.
 */

/**
 * Typography.
 * ---
 * The following properties are NOT REQUIRED by Evergreen.
 * It's exposed for convenience and documentation.
 */

/**
 * Component Specific.
 * ---
 * These ARE REQUIRED for Evergreen to work.
 */

/**
 * Theme Helpers.
 * ---
 * These ARE REQUIRED for Evergreen to work.
 */
var _default = {
  // Foundational Styles.
  colors: _foundationalStyles.colors,
  elevations: _foundationalStyles.elevations,
  fills: _foundationalStyles.fills,
  palette: _foundationalStyles.palette,
  scales: _foundationalStyles.scales,
  // Component Specific.
  avatarColors: _componentSpecific.avatarColors,
  badgeColors: _componentSpecific.badgeColors,
  spinnerColor: _componentSpecific.spinnerColor,
  overlayBackgroundColor: _componentSpecific.overlayBackgroundColor,
  getBadgeClassName: _componentSpecific.getBadgeClassName,
  getButtonClassName: _componentSpecific.getButtonClassName,
  getLinkClassName: _componentSpecific.getLinkClassName,
  getCheckboxClassName: _componentSpecific.getCheckboxClassName,
  getRadioClassName: _componentSpecific.getRadioClassName,
  getTagInputClassName: _componentSpecific.getTagInputClassName,
  getTextInputClassName: _componentSpecific.getTextInputClassName,
  getTextareaClassName: _componentSpecific.getTextareaClassName,
  getTextDropdownButtonClassName: _componentSpecific.getTextDropdownButtonClassName,
  getTabClassName: _componentSpecific.getTabClassName,
  getTableCellClassName: _componentSpecific.getTableCellClassName,
  getTooltipProps: _componentSpecific.getTooltipProps,
  getRowClassName: _componentSpecific.getRowClassName,
  getMenuItemClassName: _componentSpecific.getMenuItemClassName,
  getSelectClassName: _componentSpecific.getSelectClassName,
  getSegmentedControlRadioClassName: _componentSpecific.getSegmentedControlRadioClassName,
  getSwitchClassName: _componentSpecific.getSwitchClassName,
  getAlertProps: _componentSpecific.getAlertProps,
  getCodeProps: _componentSpecific.getCodeProps,
  getAvatarProps: _componentSpecific.getAvatarProps,
  getBadgeProps: _componentSpecific.getBadgeProps,
  getAvatarInitialsFontSize: _componentSpecific.getAvatarInitialsFontSize,
  // Theme Helpers.
  getBorderRadiusForControlHeight: _themeHelpers.getBorderRadiusForControlHeight,
  getTextSizeForControlHeight: _themeHelpers.getTextSizeForControlHeight,
  getIconSizeForButton: _themeHelpers.getIconSizeForButton,
  getIconSizeForInput: _themeHelpers.getIconSizeForInput,
  getIconSizeForSelect: _themeHelpers.getIconSizeForSelect,
  getIconSizeForIconButton: _themeHelpers.getIconSizeForIconButton,
  getBackground: _themeHelpers.getBackground,
  getElevation: _themeHelpers.getElevation,
  getIconColor: _themeHelpers.getIconColor,
  getIconForIntent: _themeHelpers.getIconForIntent,
  getHeadingStyle: _themeHelpers.getHeadingStyle,
  getTextStyle: _themeHelpers.getTextStyle,
  getParagraphStyle: _themeHelpers.getParagraphStyle,
  getFontFamily: _themeHelpers.getFontFamily,
  getTextColor: _themeHelpers.getTextColor,
  typography: {
    headings: _typography.headings,
    text: _typography.text,
    fontFamilies: _typography.fontFamilies,
    paragraph: _typography.paragraph
  }
};
exports["default"] = _default;
